src/software/multithreading/threaded_observer.tpp

```diff
@@ -45,3 +47,17 @@ ThreadedObserver<T>::~ThreadedObserver()
     // running we will segfault
     pull_from_buffer_thread.join();
 }
+
+template <typename T>
+std::optional<T> FirstInFirstOutThreadedObserver<T>::getNextValue(
+    const Duration& max_wait_time)
+{
+    return this->popLeastRecentlyReceivedValue(max_wait_time);
+}
+
+template <typename T>
+std::optional<T> LastInFirstOutThreadedObserver<T>::getNextValue(
+    const Duration& max_wait_time)
+{
+    return this->popLeastRecentlyReceivedValue(max_wait_time);
+}
```

Comment:
Both sub-classes seem to be doing the same thing? We probably want move the current unit tests to be for the `FirstInFirstOutThreadedObserver` and then basically copy them for the `LastInFirstOutThreadedObserver`?

Reply:
They only do the same thing if buffer_size == 1

Reply:
Sorry, both implementations call `popLeastRecentlyReceivedValue`? Wouldn't that do the same thing?

Reply:
fixed